backend/compact-connect/lambdas/purchases/tests/unit/test_purchase_client.py

```diff
@@ -0,0 +1,396 @@
+# ruff: noqa: ARG001 unused-argument
+import json
+from decimal import Decimal
+from unittest.mock import MagicMock, patch
+
+from exceptions import CCFailedTransactionException, CCInternalException
+
+from tests import TstLambdas
+
+MOCK_LOGIN_ID = 'mock_login_id'
+MOCK_TRANSACTION_KEY = 'R2d237rZu59q123'
+MOCK_ASLP_SECRET = {
+    'processor': 'authorize.net',
+    'api_login_id': MOCK_LOGIN_ID,
+    'transaction_key': MOCK_TRANSACTION_KEY,
+}
+
+MOCK_TRANSACTION_ID = '123456'
+
+EXPECTED_TOTAL_FEE_AMOUNT = 150.50
+
+
+def json_to_magic_mock(json_obj):
+    """
+    Create a MagicMock object from a JSON object. This is useful for mocking the authorize.net SDK response object.
+
+    Unfortunately, the authorize.net SDK response object uses attributes for all the values, which are not easily
```

Comment:
what do you mean "uses attributes for all the values"?

Reply:
The SDK uses python attributes (variables accessed by the . operator within an object) to store all of the response fields.